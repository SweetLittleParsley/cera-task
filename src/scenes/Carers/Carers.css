.carer-list {
  display: flex;
  flex-wrap: wrap;
  padding: 30px 44px;
  position: relative;
}

.carer-list__item {
  width: 25%;
  margin-bottom: 25px;
}

.carer-list__item--disabled {
  pointer-events: none;
}

.carer-list__item > div {
  padding-right: 25px;
}

.carer-list__modal {
  position: absolute;
  width: 100%;
  background-color: rgba(196, 196, 196, 0.67);
  height: 100vh;
  top: 0;
  left: 0;
}
