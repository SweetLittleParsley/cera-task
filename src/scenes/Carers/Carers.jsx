import { useEffect, useState } from 'react';
import './Carers.css';
import Carer from '../../components/Carer/Carer'
import Header from '../../components/Header/Header'
import SchedulingModal from '../../components/SchedulingModal/SchedulingModal'
import { getCarers, getAvailableSlots, bookSlot } from '../../services/data/carers'

const formatSlot = slot => {
  const time = slot.split(':')
  let date = new Date()
  date.setHours(time[0])
  date.setMinutes(time[1])
  return date.toLocaleTimeString('en-US', { timeStyle: 'short' })
}

function Carers() {
  let [carers, setCarers] = useState([])
  let [activeCarer, setActiveCarer] = useState(false)
  let [carerAvailableSlots, setCarerAvailableSlots] = useState([])
  useEffect(() => {
    async function fetchData() {
      const data = await getCarers()
      setCarers(data.carers)
    }
    fetchData()
  }, [])

  useEffect(() => {
    if (!activeCarer) return

    async function fetchData() {
      const data = await getAvailableSlots()
      const formattedSlots = data.UTCAvailableSlots.map(slot => formatSlot(slot))
      setCarerAvailableSlots(formattedSlots)
    }
    fetchData()
  }, [activeCarer])

  const scheduler = (name, slot) => {
    bookSlot({carer: name, slot})
  }

  return (
    <main>
      <Header />
      <section className={'carer-list'}>
        {carers.map(carer => {
          return (
          <article className={`carer-list__item ${activeCarer ? 'carer-list__item--disabled' : ''}`} key={carer.photo}>
            <Carer name={carer.name} availableSlots={carer.slots} image={carer.photo} onClick={() => setActiveCarer(carer.name)} />
          </article>
          )
        })}
        {activeCarer && <div className='carer-list__modal' onClick={() => setActiveCarer('')}>
          <SchedulingModal name={activeCarer} slots={carerAvailableSlots} scheduler={scheduler} />
        </div>}
      </section>
    </main>
  );
}

export default Carers;
