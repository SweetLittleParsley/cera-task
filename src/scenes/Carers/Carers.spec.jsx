import { render } from '@testing-library/react'
import Carers from './Carers'

describe('Carers', () => {
  it('renders correctly', () => {
    expect(render(<Carers />).container).toMatchSnapshot() 
  })
})