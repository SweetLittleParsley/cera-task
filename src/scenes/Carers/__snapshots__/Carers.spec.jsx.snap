// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carers renders correctly 1`] = `
<div>
  <main>
    <header
      class="header"
    >
      <img
        alt="cera logo"
        class="header__logo"
        src="cera-logo.png"
      />
      <div
        class="header__headline"
      >
        <p
          class="header__title"
        >
          Carers
        </p>
        <p
          class="header__intro"
        >
          Here you'll be able to schedule your carers visits
        </p>
      </div>
      <time
        class="header__clock"
        datetime="1632069927862"
      >
        17:45 PM
      </time>
    </header>
    <section
      class="carer-list"
    />
  </main>
</div>
`;
